be/src/olap/rowset/segment_v2/column_writer.cpp

```diff
@@ -173,6 +173,119 @@ Status ColumnWriter::create(const ColumnWriterOptions& opts, const TabletColumn*
+
+            // create null writer
+            ScalarColumnWriter* null_writer = nullptr;
+            if (opts.meta->is_nullable()) {
+                FieldType null_type = FieldType::OLAP_FIELD_TYPE_TINYINT;
+                ColumnWriterOptions null_options;
+                null_options.meta = opts.meta->add_children_columns();
+                null_options.meta->set_column_id(3);
+                null_options.meta->set_unique_id(3);
+                null_options.meta->set_type(null_type);
+                null_options.meta->set_is_nullable(false);
+                null_options.meta->set_length(
+                        get_scalar_type_info<OLAP_FIELD_TYPE_TINYINT>()->size());
+                null_options.meta->set_encoding(DEFAULT_ENCODING);
+                null_options.meta->set_compression(opts.meta->compression());
+
+                null_options.need_zone_map = false;
+                null_options.need_bloom_filter = false;
+                null_options.need_bitmap_index = false;
+
+                TabletColumn null_column = TabletColumn(
+                        OLAP_FIELD_AGGREGATION_NONE, null_type, false,
+                        null_options.meta->unique_id(), null_options.meta->length());
+                null_column.set_name("nullable");
+                null_column.set_index_length(-1); // no short key index
+                std::unique_ptr<Field> null_field(FieldFactory::create(null_column));
+                null_writer =
+                        new ScalarColumnWriter(null_options, std::move(null_field), file_writer);
+            }
+
+            // create key writer
+            std::unique_ptr<ColumnWriter> key_writer;
+	    ColumnWriterOptions key_opts;
+            TabletColumn key_list_column(OLAP_FIELD_AGGREGATION_NONE, OLAP_FIELD_TYPE_ARRAY);
+            {
+		key_list_column.add_sub_column(const_cast<TabletColumn&>(key_column));
+//                key_list_column.add_sub_column(key_column);
+                key_list_column.set_name("map.key");
+                key_list_column.set_index_length(-1);
+
+
+                key_opts.meta = opts.meta->mutable_children_columns(0);
+                key_opts.meta->set_column_id(4);
+                key_opts.meta->set_unique_id(4);
+                key_opts.meta->set_type(OLAP_FIELD_TYPE_ARRAY);
+                key_opts.meta->set_length(0);
+		key_opts.meta->set_encoding(BIT_SHUFFLE);
+                key_opts.meta->set_compression(segment_v2::CompressionTypePB::LZ4F);
+                key_opts.need_zone_map = false;
+		// no need key array's null map
+                key_opts.meta->set_is_nullable(false);
+
+                ColumnMetaPB* child_meta = key_opts.meta->add_children_columns();
+                child_meta->set_column_id(5);
+                child_meta->set_unique_id(5);
+                child_meta->set_type(key_column.type());
+                child_meta->set_length(key_column.length());
+		child_meta->set_compression(segment_v2::CompressionTypePB::LZ4F);
```

Comment:
the compression type should be same in one tablet_schema.
use key_opts.meta->compression().